### DISCUSSION TITLE
class DcmEcho needed

### DISCUSSION THREAD

**Discussion by sblour** [dcm4che/dcm4che#1095] [Q&A]
Our application built on dcm4che version 2.0.29 , to upgrade to version 3, lost class DcmEcho in dcm4che2-tool-dcmecho-2.0.29.
What is the replacement of DcmEcho class?

**Comment by hczedik** [comment]
The storescu command line tool contains the code to do a C-ECHO.
Refer to https://github.com/dcm4che/dcm4che/tree/master/dcm4che-tool/dcm4che-tool-storescu
_"If no DICOM file is specified, it sends a DICOM C-ECHO message and waits for a response."_

The code is here: https://github.com/dcm4che/dcm4che/blob/master/dcm4che-tool/dcm4che-tool-storescu/src/main/java/org/dcm4che3/tool/storescu/StoreSCU.java#L471 (method "echo()").

**Comment by sblour** [comment]
How a running application use storescu in command line?


From: Hermann Czedik-Eysenberg ***@***.***>
Sent: Monday, January 10, 2022 12:37 AM
To: dcm4che/dcm4che ***@***.***>
Cc: Blourtchian, Sousan (Suzanne), M.S. ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [dcm4che/dcm4che] class DcmEcho needed (Discussion #1095)


The storescu command line tool contains the code to do a C-ECHO.
Refer to https://github.com/dcm4che/dcm4che/tree/master/dcm4che-tool/dcm4che-tool-storescu
"If no DICOM file is specified, it sends a DICOM C-ECHO message and waits for a response."

—
Reply to this email directly, view it on GitHub<https://github.com/dcm4che/dcm4che/discussions/1095#discussioncomment-1938218>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AMALD6FCJDR6GVQXBPDFHOLUVKEA5ANCNFSM5LM6QOCA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by sblour** [comment]
We are moving to use dcm4che3. There is big shift from dcm4che2 to this version.
Where I can find replacements, guide, or document on what is remove, changed, replaced?
For example, I cannot find replacement for DicomObject that we are using in our code.
Please advise.

Thanks,
Suzanne

From: Hermann Czedik-Eysenberg ***@***.***>
Sent: Monday, January 10, 2022 12:37 AM
To: dcm4che/dcm4che ***@***.***>
Cc: Blourtchian, Sousan (Suzanne), M.S. ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [dcm4che/dcm4che] class DcmEcho needed (Discussion #1095)


The storescu command line tool contains the code to do a C-ECHO.
Refer to https://github.com/dcm4che/dcm4che/tree/master/dcm4che-tool/dcm4che-tool-storescu
"If no DICOM file is specified, it sends a DICOM C-ECHO message and waits for a response."

—
Reply to this email directly, view it on GitHub<https://github.com/dcm4che/dcm4che/discussions/1095#discussioncomment-1938218>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AMALD6FCJDR6GVQXBPDFHOLUVKEA5ANCNFSM5LM6QOCA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by SamuelImriska** [reply]
Moving to dcm4che3 (I expect you mean dcm4che -5.x) is definitely good choice, however there is no easy way of migration and to me to guide or similar.

DicomObject -> Attributes
DicomElement -> no replacement

Most of classes are similar, but the usage is more fluent.
Be careful especially on handling Sequences. Private Tags handling is more explicit.

On the positive side you gain much better handling of character sets, timezones and much smaller memory footprint.

**Comment by sblour** [comment]
Thank you for your quick reply.  We are more concerned about memory leak that we see in the old version. Hope the migration to dcm4che-5.x.x will provide better performance.

Regards,

From: Samuel Imriska ***@***.***>
Sent: Wednesday, January 26, 2022 12:58 PM
To: dcm4che/dcm4che ***@***.***>
Cc: Blourtchian, Sousan (Suzanne), M.S. ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [dcm4che/dcm4che] class DcmEcho needed (Discussion #1095)


Moving to dcm4che3 (I expect you mean dcm4che -5.x) is definitely good choice, however there is no easy way of migration and to me to guide or similar.

DicomObject -> Attributes
DicomElement -> no replacement

Most of classes are similar, but the usage is more fluent.
Be careful especially on handling Sequences. Private Tags handling is more explicit.

On the positive side you gain much better handling of character sets, timezones and much smaller memory footprint.

—
Reply to this email directly, view it on GitHub<https://github.com/dcm4che/dcm4che/discussions/1095#discussioncomment-2053625>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AMALD6BVHKDMNF3D5IU2QUDUYBG47ANCNFSM5LM6QOCA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by sblour** [comment]
What about Tags, CommandUtil classes?

Thanks,

From: Samuel Imriska ***@***.***>
Sent: Wednesday, January 26, 2022 12:58 PM
To: dcm4che/dcm4che ***@***.***>
Cc: Blourtchian, Sousan (Suzanne), M.S. ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [dcm4che/dcm4che] class DcmEcho needed (Discussion #1095)


Moving to dcm4che3 (I expect you mean dcm4che -5.x) is definitely good choice, however there is no easy way of migration and to me to guide or similar.

DicomObject -> Attributes
DicomElement -> no replacement

Most of classes are similar, but the usage is more fluent.
Be careful especially on handling Sequences. Private Tags handling is more explicit.

On the positive side you gain much better handling of character sets, timezones and much smaller memory footprint.

—
Reply to this email directly, view it on GitHub<https://github.com/dcm4che/dcm4che/discussions/1095#discussioncomment-2053625>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AMALD6BVHKDMNF3D5IU2QUDUYBG47ANCNFSM5LM6QOCA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by simriska** [comment]
Tags-> Tag
Commandutil was bunch of utility methods for network communication this is
separated into multiple respective classes
